drivers/sensor/ntc_thermistor/ntc_thermistor.c

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2023 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/adc.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/logging/log.h>
+#include "ntc_thermistor.h"
+
+LOG_MODULE_REGISTER(NTC_THERMISTOR, CONFIG_SENSOR_LOG_LEVEL);
+
+struct ntc_thermistor_data {
+	struct k_mutex mutex;
+	int16_t raw;
+	int16_t sample_val;
+};
+
+struct ntc_thermistor_config {
+	const struct adc_dt_spec adc_channel;
+	const struct ntc_config ntc_cfg;
+};
+
+static const struct ntc_compensation comp_epcos_b57861s0103a039[] = {
```

Comment:
Do we want to have tables for every possible NTC thermistor?

Also, doing it like this dictates the range and steps.

Reply:
I think it's a good idea to simplify usage. Imagine we have 10 boards with the same NTC in-tree. We'll basically need to maintain 10 copies of the compensation table. I also tend to trust Linux drivers more than Zephyr... :laughing: